### DISCUSSION TITLE
compiling error:

### DISCUSSION THREAD

**Discussion by prockscac** [KadeArchive/Kade-Engine#458] [Q&A] [answered]
cd /Users/name/Downloads/Kade-Engine-1.4.2 
name Kade-Engine-1.4.2 % lime test mac
/Users/justinspain/install lime 7.8.0/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
/Users/justinspain/install lime 7.8.0/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
source/PlayState.hx:478: characters 3-39 : Warning : `FlxCamera.defaultCameras` is deprecated, use `FlxG.cameras.setDefaultDrawTarget` instead
Creating /Users/usersname/Downloads/Kade-Engine-1.4.2/export/release/macos/obj/obj/darwin64/__pch/haxe/hxcpp.h.gch...
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
Error: Could not create PCH
usersname Kade-Engine-1.4.2 % sudo apt install gcc build-essential 
Password: 
The operation couldn’t be completed. Unable to locate a Java Runtime that supports apt.
Please visit http://www.java.com for information on installing Java.

(still happens when I go to the link and install java

.)

**Comment by puyoxyz** [comment, marked as answer]
oh im kinda really dumb i think i thought u were on a different os than u are SORRY

install xcode and try again

**Reply by prockscac** [reply]
what's x code and where do I download it?

**Reply by isakube** [reply]
https://developer.apple.com/xcode/

**Reply by prockscac** [reply]
it works! I have some other problems but ill make a slide for that.
